proposals/allow_policy_exception.md

```diff
@@ -0,0 +1,358 @@
+
+- **Policy User**: able to create, delete , and modify `PolicyExceptionRequest` resources
+- **Policy Admin**: able to review `PolicyExceptionRequest`; and create, delete, modify `PolicyExceptionApprovals` resources
+
+1. A user that is impacted by a policy violation can create a `PolicyExceptionRequest` for their workload:
+   `PolicyExceptionRequest` sample:
+
+```yaml
+spec:
+  exceptions:
+  - policyName: "pod-security"
+    ruleNames:
+    - "restricted"
+    reason: "..."
+    exclude: {}      # same as a policy exclude declaration
+    duration: "30d"  # optional time string 
+status:
+  state: "Pending" # Pending | Approved | Denied
+```
+
+`spec` is where the user fills in the request.
+
+- exceptions: a list of exception objects:
+  - policyName: target policy name (string)
+  - ruleNames: an array of target rule names (string)
+  - reason: brief explanation for the exception
+  - exclude: specify the resource that need to be excluded from the target policy, it can be [MatchResources](https://github.com/kyverno/kyverno/blob/v1.8.0-rc2/api/kyverno/v1/match_resources_types.go#L10)
+  - duration(optional): how long the exception would be valid
+
+`status` is where the `PolicyExceptionController` would automatically update based on Policy Admin activity and other updates.
+
+**PolicyExceptionRequest lifecycle**
+
+* If the `state` is `Pending` (the default), user can delete or modify the PolicyExceptionRequest. 
+* If the state is `Approved`, user will be blocked from modifying the PolicyExceptionRequest. 
+* The user can always delete the PolicyExceptionRequest when it is no longer required. 
+* If the state is `Denied`, user can modify the request. The controller would change the state back to `Pending`. We can use RBAC to restrict such accessibility.
+
+2. The admin can review the `PolicyExceptionRequest` and can either `approve` or `deny` it. This is done by updating the `PolicyExceptionApprovals` which serves as a log of approval (or denial) decisions.
+
+`PolicyExceptionApproval` sample:
+
+```yaml
+spec:
+  approvalPolicy: Manual              # Manual (default) | Automatic
+  exceptions:
+    - name: permit-root-user
+      namespace: nginx
+      status: "Approved"
+      reason: "..."
+    - name: permit-host-port
+      namespace: nginx
+      status: "Denied"
+      reason: ...
+```
+
+- approvalPolicy: policies that are approved manually / automatically
+- exceptions: an array of exception elements
+  - name: name of corresponding `PolicyExceptionRequest`
+  - namespace: namespace of corresponding `PolicyExceptionRequest`
+  - status: whether admin decide to approve / deny the `PolicyExceptionRequest`
+  - reason: why admin approve / deny the `PolicyExceptionRequest`
+
+The `status` and `reason` would also be updated to the `PolicyExceptionRequest` to let the user know by the `PolicyExceptionController`.
+
+### Ideal Usage
+
+1. User creates a CR `PolicyExceptionRequest` to request for exception in a certain namespace.
```

Comment:
The PER is a namespaced resource which makes sense, but two things we should consider:

1. How or if a user may request an exception for a cluster-scoped resource (ex., a Calico GlobalNetworkPolicy)
2. Protection so that a user in Namespace A cannot create an exception for a violating resource in Namespace B.

Reply:
Sounds good 👍 We update the KDP, now the PER can be either namespaced / cluster-wide.